modules/auxiliary/gather/enum_dns.rb

```diff
@@ -1,483 +1,422 @@
-                when "TXT"
-                  print_status("Text: #{rr.inspect}")
-                when "SRV"
-                  print_status("Host: #{rr.host} Port: #{rr.port} Priority: #{rr.priority} Record: SRV")
-                end
-                rescue ActiveRecord::RecordInvalid
-                  # Do nothing. Probably tried to store :host => 127.0.0.1
-                end
-              end
-            end
-          else
-            print_error("Zone transfer failed (length was zero)")
-          end
-        rescue Exception => e
-          print_error("Error executing zone transfer: #{e.message}")
-          elog("Error executing zone transfer: #{e.message}\n#{e.backtrace.join("\n")}")
-        end
-      end
+  def get_mx(domain)
+    resp = dns_query(domain, 'MX')
+    return if resp.blank? || resp.answer.blank?
 
-    else
-      print_error("Could not resolve domain #{target}")
+    records = []
+    resp.answer.each do |r|
+      next unless r.class == Net::DNS::RR::MX
+      records << "#{r.exchange}"
+      report_host(host: r.exchange, name: domain, info: 'MX')
+      vprint_good("#{domain}: MX: #{r.exchange}")
     end
+    return if records.none?
+    save_loot('ENUM_MX', 'text/plain', domain, "#{records.join(',')}", domain)
+    records
   end
 
-  def run
-    @res = Net::DNS::Resolver.new()
-    if datastore['TCP_DNS']
-      vprint_status("Using DNS/TCP")
-      @res.use_tcp = true
-    end
-    @res.retry = datastore['RETRY'].to_i
-    @res.retry_interval = datastore['RETRY_INTERVAL'].to_i
-    @threadnum = datastore['THREADS'].to_i
-    wldcrd = wildcard(datastore['DOMAIN'])
-    switchdns(datastore['DOMAIN'])
-
-    if(datastore['ENUM_STD'])
-      genrcd(datastore['DOMAIN'])
+  def get_soa(domain)
+    resp = dns_query(domain, 'SOA')
+    return if resp.blank? || resp.answer.blank?
+
+    records = []
+    resp.answer.each do |r|
+      next unless r.class == Net::DNS::RR::SOA
+      records << r.mname
+      report_host(host: r.mname, info: 'SOA')
+      vprint_good("#{domain}: SOA: #{r.mname}")
     end
+    return if records.none?
+    save_loot('ENUM_SOA', 'text/plain', domain, "#{records.join(',')}", domain)
+    records
+  end
+
+  def get_txt(domain)
+    resp = dns_query(domain, 'TXT')
+    return if resp.blank? || resp.answer.blank?
 
-    if(datastore['ENUM_TLD'])
-      tldexpnd(datastore['DOMAIN'],datastore['NS'])
+    records = []
+    resp.answer.each do |r|
+      next unless r.class == Net::DNS::RR::TXT
+      records << r.txt
+      vprint_good("#{domain}: TXT: #{r.txt}")
     end
+    return if records.none?
+    save_loot('ENUM_TXT', 'text/plain', domain, "#{records.join(',')}", domain)
+    records
+  end
 
-    if(datastore['ENUM_BRT'])
-      if wldcrd and datastore['STOP_WLDCRD']
-        print_error("Wildcard record found!")
-      else
-        dnsbrute(datastore['DOMAIN'],datastore['WORDLIST'],datastore['NS'])
-      end
+  def get_tld(domain)
+    domain_ = domain.split('.')
```

Comment:
Especially because this is now run by default, I would recommend adding a `print` early on in this function to indicate that the TLD checking is occurring and that it can take some time.